app/lib/actions/katello/content_view/incremental_updates.rb

```diff
@@ -29,14 +35,40 @@ def plan(version_environments, composite_version_environments, content, dep_solv
             end
 
             if hosts.any? && !content[:errata_ids].blank?
-              errata = ::Katello::Erratum.with_identifiers(content[:errata_ids])
-              hosts = hosts.where(:id => ::Katello::Host::ContentFacet.with_applicable_errata(errata).pluck(:host_id))
               plan_action(::Actions::BulkAction, ::Actions::Katello::Host::Erratum::ApplicableErrataInstall, hosts, content[:errata_ids])
             end
             plan_self(:version_outputs => output_for_version_ids)
           end
         end
 
+        def find_applicable_hosts_and_smart_proxies(hosts, content)
+          smart_proxies = []
+          if hosts.any? && !content[:errata_ids].blank?
+            errata = ::Katello::Erratum.with_identifiers(content[:errata_ids])
+            hosts = hosts.where(:id => ::Katello::Host::ContentFacet.with_applicable_errata(errata).pluck(:host_id))
+            smart_proxies = (hosts.map { |host| host.subscription_facet_attributes.registered_through unless host.subscription_facet_attributes.nil? }).uniq
+            smart_proxies = (smart_proxies.map { |proxy| SmartProxy.find_by_name(proxy) unless proxy.nil? || SmartProxy.find_by_name(proxy).default_capsule? }).compact
```

Comment:
This could be re-written as:  
```
smart_proxies = SmartProxy.where(:name => smart_proxies)
```
I'm not sure we need to include the default capsule here?

Reply:
Good point; while `hosts` must contain also hosts under default Capsule (and they do), `smart_proxies` shouldnt contain the default one.